chat_shell/chat_shell/tools/skill_factory.py

```
@@ -295,13 +322,95 @@ async def prepare_skill_tools(
+                    task_id,
+                )
+            else:
+                logger.warning(
+                    "[skill_factory] Failed to connect to skill MCP servers for task %d",
+                    task_id,
+                )
+        except asyncio.TimeoutError:
+            logger.error(
+                "[skill_factory] Timeout connecting to skill MCP servers for task %d",
+                task_id,
+            )
```

Ruff (0.14.14)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)